exporter/elasticsearchexporter/model.go

```diff
@@ -41,16 +41,65 @@ const (
 
 func (m *encodeModel) encodeLog(resource pcommon.Resource, record plog.LogRecord, scope pcommon.InstrumentationScope) ([]byte, error) {
 	var document objmodel.Document
-	document.AddTimestamp("@timestamp", record.Timestamp()) // We use @timestamp in order to ensure that we can index if the default data stream logs template is used.
-	document.AddTraceID("TraceId", record.TraceID())
-	document.AddSpanID("SpanId", record.SpanID())
-	document.AddInt("TraceFlags", int64(record.Flags()))
-	document.AddString("SeverityText", record.SeverityText())
-	document.AddInt("SeverityNumber", int64(record.SeverityNumber()))
-	document.AddAttribute("Body", record.Body())
-	m.encodeAttributes(&document, record.Attributes())
-	document.AddAttributes("Resource", resource.Attributes())
-	document.AddAttributes("Scope", scopeToAttributes(scope))
+
+	switch m.mode {
+	case MappingECS:
+		document.AddTimestamp("@timestamp", record.Timestamp()) // We use @timestamp in order to ensure that we can index if the default data stream logs template is used.
+		document.AddTraceID("trace.id", record.TraceID())
+		document.AddSpanID("span.id", record.SpanID())
+
+		if f := record.Flags(); f != 0 {
+			document.AddInt("trace.flags", int64(record.Flags()))
+		}
+
+		if n := record.SeverityNumber(); n != plog.SeverityNumberUnspecified {
+			document.AddInt("log.syslog.severity.code", int64(record.SeverityNumber()))
+			document.AddString("log.syslog.severity.name", record.SeverityText())
+		}
+
+		document.AddAttribute("message", record.Body())
```

Comment:
`record.Body()` could be a scalar or complex value, as documented in https://github.com/open-telemetry/opentelemetry-proto/blob/9d139c87b52669a3e2825b835dd828b57a455a55/opentelemetry/proto/logs/v1/logs.proto#L167-L170.  

>   // A value containing the body of the log record. Can be for example a human-readable
  // string message (including multi-line) describing the event in a free form or it can
  // be a structured data composed of arrays and maps of other values. [Optional].
  opentelemetry.proto.common.v1.AnyValue body = 5;

So I think this conversion needs to handle those different possibilities.  Alternatively, for now, you could just check if `record.Body().Type()` is `pcommon.ValueTypeStr` and if so set `message` to `record.Body().Str()`.  We can handle the other possibilities in a follow up PR.